### ISSUE TITLE
io.trino.testing.QueryFailedException: cannot unset noMoreSplits

### ISSUE THREAD

**Issue by arhimondr** [trinodb/trino#15931]
```
Error:  io.trino.plugin.mysql.TestMySqlTaskFailureRecoveryTest.testParallel[testDelete](2)  Time elapsed: 3.956 s  <<< FAILURE!
io.trino.testing.QueryFailedException: cannot unset noMoreSplits
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:490)
	at io.trino.testing.BaseFailureRecoveryTest$FailureRecoveryAssert.lambda$execute$2(BaseFailureRecoveryTest.java:483)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at io.trino.testing.BaseFailureRecoveryTest$FailureRecoveryAssert.execute(BaseFailureRecoveryTest.java:483)
	at io.trino.testing.BaseFailureRecoveryTest$FailureRecoveryAssert.executeExpected(BaseFailureRecoveryTest.java:449)
	at io.trino.testing.BaseFailureRecoveryTest$FailureRecoveryAssert.isCoordinatorOnly(BaseFailureRecoveryTest.java:527)
	at io.trino.plugin.jdbc.BaseJdbcFailureRecoveryTest.testDelete(BaseJdbcFailureRecoveryTest.java:52)
	at io.trino.testing.BaseFailureRecoveryTest$ParallelTestRunnable.run(BaseFailureRecoveryTest.java:806)
	at io.trino.testing.BaseFailureRecoveryTest.testParallel(BaseFailureRecoveryTest.java:200)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:75)
	at org.testng.internal.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:14)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
	Suppressed: java.lang.Exception: SQL: CREATE TABLE table_5ls7fm7l8c AS SELECT * FROM orders
		at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:493)
		... 23 more
Caused by: com.google.common.base.VerifyException: cannot unset noMoreSplits
	at com.google.common.base.Verify.verify(Verify.java:126)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunnerFactory.enqueueSplits(SqlTaskExecution.java:618)
	at io.trino.execution.SqlTaskExecution.updateSplitAssignments(SqlTaskExecution.java:273)
	at io.trino.execution.SqlTaskExecution.addSplitAssignments(SqlTaskExecution.java:222)
	at io.trino.execution.SqlTask.updateTask(SqlTask.java:460)
	at io.trino.execution.SqlTaskManager.doUpdateTask(SqlTaskManager.java:500)

```

**Comment by arhimondr** [comment]
https://github.com/trinodb/trino/actions/runs/4059500170/jobs/6989283854

**Comment by arhimondr** [comment]
This problem used to happen before due to duplicated `SplitAssignment` not being filtered out:

https://github.com/trinodb/trino/issues/14656

Was fixed by https://github.com/trinodb/trino/pull/14668

The issue started reoccuring recently, could potentially be caused by changes introduced in https://github.com/trinodb/trino/pull/15721

@JunhyungSong Could you please take a look?

**Comment by arhimondr** [comment]
I believe this [check](https://github.com/trinodb/trino/pull/15721/files#diff-29eb8c360b3f8c5c2604c2c71e1c291989b5655f26c009f9a38c6ad89d96fe34R242) has to be applied after filtering out duplicated splits by sequence id

**Comment by arhimondr** [comment]
It looks like @pettyjamesm ran into this issue and created a fix. We should probably extract it in a separate PR and merge it to unblock the release: https://github.com/trinodb/trino/pull/15917/commits/d2084676b4b2270a593a702f65919b10ddd9418a

**Comment by JunhyungSong** [comment]
@pettyjamesm will separate the commit out of https://github.com/trinodb/trino/pull/15917.

**Comment by pettyjamesm** [comment]
Opened https://github.com/trinodb/trino/pull/15933 to address it.
